pytest_ethereum/_utils/linker.py

```diff
@@ -7,35 +7,40 @@
 from ethpm import Package
 from ethpm.typing import URI, Address, Manifest
 from ethpm.utils.chains import (
-    check_if_chain_matches_chain_uri,
     create_block_uri,
     get_genesis_block_hash,
+    parse_BIP122_uri,
 )
 from pytest_ethereum.exceptions import LinkerError
 from pytest_ethereum.typing import TxReceipt
 
 
-def pluck_matching_uri(deployment_data: Dict[str, Dict[str, str]], w3: Web3) -> URI:
+def pluck_matching_uri(deployment_data: Dict[str, Dict[str, str]], new_uri: URI) -> URI:
     """
     Return any blockchain uri that matches w3-connected chain, if one
     is present in the deployment data keys.
     """
+    new_genesis, _, _ = parse_BIP122_uri(new_uri)
     for uri in deployment_data.keys():
-        if check_if_chain_matches_chain_uri(w3, uri):
+        old_genesis, _, _ = parse_BIP122_uri(uri)
+        if new_genesis == old_genesis:
```

Comment:
Simply matching genesis hash doesn't guarantee that the chains match.  The ETC chain has the same genesis as the ETH chain.  I believe you **must** have a `Web3` instance to do this matching because otherwise you can't know that the two URIs are on the same chain.